### PR TITLE
Switch to Yarn from Bower

### LINKED ISSUE #351: Switch from Bower to Yarn for JavaScript dependencies
We did this in `plots2` and it took a while to figure out, but ultimately wasn't too complex:

https://github.com/publiclab/plots2/pull/2659/files

Let's do this here too!

### PR DESCRIPTION
**Pull request by kaustubh-nair**
Fixes #351 

Things to note:
1. Used the scriptaculous tar from github instead of the zip that was being used with bower, since yarn doesn't support zip files. 
2. I'm getting 3 peer dependencies warnings which I'm not sure how to fix. Can anyone help me out with this?

`warning "leaflet-environmental-layers > leaflet-spin@1.1.0" has incorrect peer dependency "leaflet@^0.7.3".`

`warning " > leaflet-illustrate@0.0.2" has unmet peer dependency "grunt-cli@0.1.13".`

`warning " > leaflet-spin@1.0.0" has incorrect peer dependency "leaflet@^0.7.3".`

Thanks! :)

### CODE REVIEW

**Comment by jywarren** [comment]
Wow this is fantastic! Does it work well? Can you share some screenshots? Thanks!!!!

**Comment by jywarren** [comment]
This is so great, btw. @icarito if you're able to set up a testing container soon we can try this out?

**Comment by icarito** [comment]
Will do! Testing out https://github.com/publiclab/mapknitter/pull/363 now!

**Comment by kaunair** [comment]
Hey! Sorry for the late reply, I was a bit caught up with exams.
I'm not sure why but the assets won't load
I made sure the assets were installed with `yarn --check`, which ran successfully.
Also tried running the docker container but assets won't load for them as well.
Any ideas how to find what's wrong?

**Comment by kaunair** [comment]
@icarito I can't install assets using bower as well when I'm using docker
I also tried changing the environment to development in docker-compose but still no use..

Here are logs. Is it a problem only with font awesome?
Thanks!

`| 
web_1  | Started GET "/lib/fontawesome/css/font-awesome.min.css" for 172.17.0.1 at 2019-03-03 17:50:56 +0000
web_1  | 
web_1  | ActionController::RoutingError (No route matches [GET] "/lib/fontawesome/css/font-awesome.min.css"):
web_1  |   actionpack (3.2.22.5) lib/action_dispatch/middleware/debug_exceptions.rb:21:in `call'
web_1  |   actionpack (3.2.22.5) lib/action_dispatch/middleware/show_exceptions.rb:56:in `call'
web_1  |   railties (3.2.22.5) lib/rails/rack/logger.rb:32:in `call_app'
web_1  |   railties (3.2.22.5) lib/rails/rack/logger.rb:16:in `block in call'
web_1  |   activesupport (3.2.22.5) lib/active_support/tagged_logging.rb:22:in `tagged'
web_1  |   railties (3.2.22.5) lib/rails/rack/logger.rb:16:in `call'
web_1  |   actionpack (3.2.22.5) lib/action_dispatch/middleware/request_id.rb:22:in `call'
web_1  |   rack (1.4.7) lib/rack/methodoverride.rb:21:in `call'
web_1  |   rack (1.4.7) lib/rack/runtime.rb:17:in `call'
web_1  |   activesupport (3.2.22.5) lib/active_support/cache/strategy/local_cache.rb:72:in `call'
web_1  |   rack (1.4.7) lib/rack/lock.rb:15:in `call'
web_1  |   actionpack (3.2.22.5) lib/action_dispatch/middleware/static.rb:83:in `call'
web_1  |   railties (3.2.22.5) lib/rails/engine.rb:484:in `call'
web_1  |   railties (3.2.22.5) lib/rails/application.rb:231:in `call'
web_1  |   rack (1.4.7) lib/rack/content_length.rb:14:in `call'
web_1  |   railties (3.2.22.5) lib/rails/rack/log_tailer.rb:17:in `call'
web_1  |   rack (1.4.7) lib/rack/handler/webrick.rb:59:in `service'
web_1  |   /usr/local/lib/ruby/2.4.0/webrick/httpserver.rb:140:in `service'
web_1  |   /usr/local/lib/ruby/2.4.0/webrick/httpserver.rb:96:in `run'
web_1  |   /usr/local/lib/ruby/2.4.0/webrick/server.rb:308:in `block in start_thread'
web_1  | 
`

**Comment by alaxalves** [comment]
@kaunair @jywarren @icarito Do you want me to merge this with #363 ? I could give it a try too.

File: .yarnrc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+--install.modules-folder "./public/lib"
+ignore-engines true
+ignore-scritps true
```
Comment (alaxalves) [inline .yarnrc]: Small typo here

**Comment by kaunair** [comment]
Sure @alaxalves! 
I'm kinda stuck here, so a bit of help would be appreciated!
Thanks

**Comment by alaxalves** [comment]
@kaunair Just successfully installed dependencies using Yarn in c04c5dd.

**Comment by jywarren** [comment]
Hey i just wanted to chime in that this is such a great collaboration!!! 👍 ❤️

**Comment by alaxalves** [comment]
@icarito Do you think it's a good idea merging this now?

**Comment by jywarren** [comment]
Hi @alaxalves -- OK, i /believe/ i've resolved the conflicts, and this should squash when we merge it -- does it look right?

**Comment by alaxalves** [comment]
@jywarren I don't think its ok, if you check the [Dockerfile](https://github.com/publiclab/mapknitter/blob/a8c28190b42bafd012be0255017be04216889e31/Dockerfile) for instance its not right.

**Comment by jywarren** [comment]
Hmm, perhaps @kaunair can try to do a clean rebase? Sorry, i may have
missed something!

**Comment by kaunair** [comment]
I've rebased and fixed conflicts, but I'm still getting

`web    | Started GET "/assets/application.js?body=1" for 172.22.0.1 at 2019-03-26 11:20:45 +0000
web    | Error compiling asset application.js:
web    | Sprockets::FileNotFound: couldn't find file 'jquery/dist/jquery.js'
web    |   (in /app/app/assets/javascripts/application.js:14)
web    | Served asset /application.js - 500 Internal Server Error
`
I think there's something wrong with the asset location?

**Comment by kaustubh-nair** [comment]
https://github.com/publiclab/mapknitter/pull/678

**Pull request closed without merging**